src/common/async-job-runner.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { hasText, isNonEmptyObject, isValidUUID } from '@adobe/spacecat-shared-utils';
+import { Audit as AuditModel } from '@adobe/spacecat-shared-data-access';
+import { ok } from '@adobe/spacecat-shared-http-utils';
+import { StepAudit } from './step-audit.js';
+import { sendContinuationMessage, isAuditEnabledForSite } from './audit-utils.js';
+
+const { AUDIT_STEP_DESTINATION_CONFIGS } = AuditModel;
+
+export class AsyncJobRunner extends StepAudit {
+  constructor(
+    jobProvider,
+    siteProvider,
+    orgProvider,
+    urlResolver,
+    persister,
+    messageSender,
+    postProcessors,
+    steps = {},
+  ) {
+    super(siteProvider, orgProvider, urlResolver, persister, messageSender, postProcessors, steps);
+
+    this.jobProvider = jobProvider;
+  }
+
+  async chainStep(step, stepResult, context) {
+    const {
+      type, job, urls, log,
```

Comment:
urls may be contained in the job, possible remove as sepearate destructuring item